`error[E9999]: assertion failed: !adt_def.is_union() || *CORE_EXTRACTION_MODE`

When running `cargo hax json` for `memchr` I get:
```
Compiling memchr v2.7.4 (/home/maxime/test/memchr)
disabled backtrace
error[E9999]: assertion failed: !adt_def.is_union() || *CORE_EXTRACTION_MODE
  --> src/memmem/searcher.rs:44:5
   |
44 | /     pub(crate) fn new<R: HeuristicFrequencyRank>(
45 | |         prefilter: PrefilterConfig,
46 | |         ranker: R,
47 | |         needle: &[u8],
48 | |     ) -> Searcher {
   | |_________________^
   |
   = note: ⚠️ This is a bug in Hax's frontend.
           Please report this error to https://github.com/hacspec/hax/issues with some context (e.g. the current crate)!

error: could not compile `memchr` (lib) due to 1 previous error
warning: hax: running `cargo build` was not successful, continuing anyway.
```

--- a/frontend/exporter/src/rustc_utils.rs
+++ b/frontend/exporter/src/rustc_utils.rs
@@ -26,7 +26,6 @@ pub(crate) fn get_variant_information<'s, S: UnderOwnerState<'s>>(
     variant_index: rustc_target::abi::VariantIdx,
     s: &S,
 ) -> VariantInformations {
-    s_assert!(s, !adt_def.is_union() || *CORE_EXTRACTION_MODE);
     fn is_named<'s, I: std::iter::Iterator<Item = &'s ty::FieldDef> + Clone>(it: I) -> bool {
         it.clone()
             .any(|field| field.name.to_ident_string().parse::<u64>().is_err())

